### PR TITLE
new way for configuration

### PR DESCRIPTION
**Pull request by cake17**
@Xety let me know what you think of this. I'll change the way the `lang` is treated (as discussed in https://github.com/cakephp-fr/recaptcha/issues/2) in another PR as soon as this PR is merged.

- This breaks the old configs, I changed the way the plugin was configured, it's now in the `app.php` and not in a `recaptcha.php` file. I think the need to create a config file is not necessary and the use of the global config is enough.
- The validation is divided into 3 validation files:
    - one GlobalValidation parent class with rules for `lang`, `type`, `size` and `theme` params : these rules should be applied everywhere.
    - one ConfigValidation child class : used in the `bootstrap.php`. This class will check Global rules (4 params) AND the secret param
    - one RecaptchaValidation child class : used in the Recaptcha helper. This class will check the Global rules (4 params)
- I added the `size` type : `normal` or `compact`.
- new langs were added by Google for Recaptcha, so the accepted lang are updated.

### CODE REVIEW

File: config/bootstrap.php [hunk 1]
```diff
@@ -7,13 +7,13 @@
...
-use Cake\Core\Configure\Engine\PhpConfig;
 use Recaptcha\Validation\ConfigValidator;
 
-// Pass the config data from config/recaptcha.php to Configure Class
-// If the file does not exist, an exception is thrown
-Configure::config('default', new PhpConfig(dirname(APP) . DS . 'config' . DS));
-Configure::load('recaptcha', 'default', false);
+$config = Configure::read('Recaptcha');
+if ($config == null) {
+    throw new \Exception(__d('recaptcha', 'Please add a configuration for the plugin in the app.php file'));
```
Comment (Xety) [inline config/bootstrap.php]: Maybe add the name `reCaptcha` in the phrase, no ? It will be more explicit i think.

Comment (cake17) [inline config/bootstrap.php]: Yes this is more explicit with recaptcha

**Comment by vaughany** [comment]
:+1: I would prefer configuration in app.php and not a separate file. I tend to keep all the secret configuration in app.php (which is not committed to git) and any more general configuration in bootstrap.php.

Also, great plugin! Got started really easily and quickly with Google's reCAPTCHA thanks to it, so thanks for the hard work. :)

**Comment by cake17** [comment]
Thanks for your comment ! :smile: I'll merge this and create a 0.4 tag so that people can continue to use the 0.3.3 version with old configuration style.

**Comment by vaughany** [comment]
:clap: :clap: :clap: Thank you! :)

**Comment by cake17** [comment]
:smile: I just wonder if I should not have let the old config style enable to avoid BC breaks... (and keeping the new way of configuration of course).
I made small updates, I'll make a new tag 0.4.1 in the near future. If you find any issue with the plugin, please let me know !! :)

**Comment by vaughany** [comment]
Will do, and thanks again.

**Pull request merged**